_drafts/zio-fibers.md

```diff
@@ -0,0 +1,290 @@
+---
+title: "ZIO and the Fiber Model"
+date: 2021-06-31 
+header:
+  image: "/images/blog cover.jpg"
+tags: []
+excerpt: "Many libraries implement the effect pattern in the Scala ecosystem, and every one has its own concurrency model. First, let's introduce the ZIO library and its implementation of the fiber model."
+---
+
+Many libraries implement the effect pattern in the Scala ecosystem: Cats Effect, Monix, and ZIO, just to list some. Every of these implements its own concurrency model. For example. Cats Effect and ZIO both rely on _fibers_. In the articles [Cats Effect 3 - Introduction to Fibers](https://blog.rockthejvm.com/cats-effect-fibers/) and [Cats Effect 3 - Racing IOs](https://blog.rockthejvm.com/cats-effect-racing-fibers/), we introduced the fiber model adopted by the Cats Effect library. Now, it's time to analyze the ZIO library and its implementation of the fiber model.
+
+## 1. Background and Setup
+
+We live in a beautiful world where Scala 3 is the actual major release of our loved programming language. So, we will use Scala 3 throughout the article. Moreover, we will need the dependency from the ZIO library:
+
+```sbt
+libraryDependencies += "dev.zio" %% "zio" % "1.0.9"
+```
+
+Nothing else will be required.
+
+## 2. The Effect Pattern
+
+Before talking about ZIO fibers, we need to have the notion of the _Effect Pattern_. The main objective of any effect library is to deal with statements that can lead to side effects. The _substitution model_ is the fundamental building block of functional programming. Unfortunately, **we cannot model side effects using the substitution model because we cannot substitute functions with  their results**, varying from information other than simple functions' inputs. Such functions are often called _impure_:
```

Comment:
with (single space) their results
What do you mean by "we cannot model..."? I think you need to be careful with the phrasing here.